packages/channels/src/telegram/api.ts

```diff
@@ -0,0 +1,97 @@
+import { Response } from 'express'
+import { Context, NarrowedContext } from 'telegraf'
+import { Update } from 'telegraf/typings/core/types/typegram'
+import yn from 'yn'
+import { ChannelApi, ChannelApiManager, ChannelApiRequest } from '../base/api'
+import { ChannelInitializeEvent, ChannelStartEvent } from '../base/service'
+import { POSTBACK_PREFIX, SAY_PREFIX } from './renderers/carousel'
+import { TelegramService } from './service'
+
+export class TelegramApi extends ChannelApi<TelegramService> {
+  async setup(router: ChannelApiManager) {
+    router.post('/telegram/:token', this.handleRequest.bind(this))
+
+    this.service.on('start', this.handleStart.bind(this))
+    this.service.on('initialize', this.handleInitialize.bind(this))
+  }
+
+  private async handleRequest(req: ChannelApiRequest, res: Response) {
+    const { config, callback } = this.service.get(req.scope)
+
+    if (req.params.token === config.botToken) {
+      req.url = '/' // by-passes verification in telegraf since we do it here instead
+      callback!(req, res)
+    } else {
+      res.sendStatus(401)
+    }
+  }
+
+  private async handleInitialize({ scope }: ChannelInitializeEvent) {
+    if (this.useWebhook()) {
+      const { telegraf, config } = this.service.get(scope)
+      const webhook = `${await this.urlCallback!(scope)}/${config.botToken}`
+      await telegraf.telegram.setWebhook(webhook)
+    }
+  }
+
+  private async handleStart({ scope }: ChannelStartEvent) {
+    const { telegraf } = this.service.get(scope)
+
+    telegraf.on('message', this.asyncCallback(scope, this.handleTelegrafMessage.bind(this)))
+    telegraf.on('callback_query', this.asyncCallback(scope, this.handleTelegrafCallbackQuery.bind(this)))
+
+    if (this.useWebhook()) {
+      this.service.get(scope).callback = telegraf.webhookCallback('/')
+    } else {
+      await telegraf.telegram.deleteWebhook()
+      await telegraf.launch()
+    }
+  }
+
+  private async handleTelegrafMessage(scope: string, ctx: NarrowedContext<Context<Update>, Update.MessageUpdate>) {
+    if ('text' in ctx.message) {
+      await this.service.receive(scope, this.extractEndpoint(ctx), { type: 'text', text: ctx.message.text })
+    }
+  }
+
+  private async handleTelegrafCallbackQuery(
+    scope: string,
+    ctx: NarrowedContext<Context<Update>, Update.CallbackQueryUpdate>
+  ) {
+    if ('data' in ctx.callbackQuery) {
+      const data = ctx.callbackQuery.data
```

Comment:
As I think you've seen as well that's the suggested way of making the typings work as per telegraf documentation. It's a bit strange but whatever